### PR TITLE
remove all occurences of config templates from robottelo

### PR DESCRIPTION
**Pull request by tstrych**
I remove all occurences of `config_template`, `ConfigTemplate` and `config-template`. 
Config templates were removed already from nailgun. Look at [PR 724](https://github.com/SatelliteQE/nailgun/pull/724/)
There is still pending [PR 726, cherry-pick of 724](https://github.com/SatelliteQE/nailgun/pull/726) to stable branch. 
When I tried to tests these changes I find out there is a problem with audit_comment, which is already solved by [PR 723](https://github.com/SatelliteQE/nailgun/pull/723/) not merge yet.

After above mentioned PR's  are merged this can be merged too. 
This should fix another bunch of tests in sat 6.8. 

I marked all tests, where changes were made and then I ran these tests locally:
(This marks were removed before this PR.)
Results:
```
tests/foreman/api/test_discoveredhost.py::DiscoveryTestCase::test_positive_upload_facts SKIPPED                                                                       [  8%]
tests/foreman/api/test_location.py::LocationTestCase::test_positive_update_entities PASSED                                                                            [ 16%]
tests/foreman/api/test_multiple_paths.py::EntityTestCase::test_positive_get_status_code PASSED                                                                        [ 25%]
tests/foreman/api/test_multiple_paths.py::EntityTestCase::test_positive_post_status_code FAILED                                                                       [ 33%]
tests/foreman/api/test_operatingsystem.py::OperatingSystemTestCase::test_positive_create_with_template PASSED                                                         [ 41%]
tests/foreman/api/test_operatingsystem.py::OperatingSystemTestCase::test_positive_update_template PASSED                                                              [ 50%]
tests/foreman/api/test_template_combination.py::TemplateCombinationTestCase::test_positive_get_combination PASSED                                                     [ 58%]
tests/foreman/cli/test_location.py::LocationTestCase::test_positive_create_update_delete PASSED                                                                       [ 66%]
tests/foreman/cli/test_location.py::LocationTestCase::test_positive_create_with_parent PASSED                                                                         [ 75%]
tests/foreman/cli/test_operatingsystem.py::OperatingSystemTestCase::test_positive_add_template PASSED                                                                 [ 83%]
tests/foreman/cli/test_organization.py::OrganizationTestCase::test_positive_add_and_remove_templates FAILED                                                           [ 91%]
tests/foreman/ui/test_host.py::test_positive_provision_end_to_end ERROR                                                                                               [100%]
================================================= 2 failed, 8 passed, 7 skipped, 2699 deselected, 1 error in 236.74 seconds =================================================
```

Explanation of tests with statuses: skipped, error or failures:
`test_discoveredhost` - skipped as vlan_networking was not setup
`test_positive_provision_end_to_end` - error as I had no infrastructure for this tests
`test_positive_add_and_remove_templates` - with help of @rplevka this was recognized as bug 
`test_positive_post_status_code` - this test was failing even before

### CODE REVIEW

File: robottelo/cli/factory.py [hunk 1]
```diff
@@ -612,6 +606,8 @@ def make_location(options=None):
         --realms REALM_NAMES                        Realm names
                                                     Comma separated list of
                                                     values.
+        --provisioning-template[s|-ids]             Provisioning template Names/Ids
```
Comment (tstrych) [inline robottelo/cli/factory.py:609]: These changes are weird a bit as this is the new way how this is written in help messages.

**Comment by tstrych** [comment]
Thanks @mirekdlugosz , I forget to look search for `config template`. I will fix that. I will learn the way you searched for it.  
I agree `tests/foreman/data/hammer_commands.json` is a bit problematic. 

I thinks it is possible, but what are the ways how can I choose the tests from file and also from different directories and run them ?

I was using 
`pytest -m removect`
But maybe  
`pytest -k "test1 or test2 or test3" can work but that will match also the test which have same test nam and are from different directories, but that should not be much overload I think :) 

There has to be better way, I just probably do not know it.  Can @mirekdlugosz  or any of you please let me know how should I do that ? 
thanks in advance.

**Comment by tstrych** [comment]
TLDR:Pending for cherry pick in nailgun

To test this via satellite6-standalone-automation cherry pick to stable branch of [nailgun PR 723](https://github.com/SatelliteQE/nailgun/pull/723/) has to be created. 

Standalone automation results now:
https://satellite-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/Satellite%206/job/satellite6-standalone-automation/1946/consoleFull

**Comment by tstrych** [comment]
the results of [satellite6-standalone-automation](https://satellite-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/Satellite%206/job/satellite6-standalone-automation/1949/)
Results are from 6.8 snap 1. 
```
============================= test session starts ==============================
collected 10 items

tests/foreman/api/test_location.py::LocationTestCase::test_positive_update_entities PASSED [ 10%]
tests/foreman/api/test_multiple_paths.py::EntityTestCase::test_positive_get_status_code PASSED [ 20%]
tests/foreman/api/test_multiple_paths.py::EntityTestCase::test_positive_post_status_code FAILED [ 30%]
tests/foreman/api/test_operatingsystem.py::OperatingSystemTestCase::test_positive_create_with_template PASSED [ 40%]
tests/foreman/api/test_operatingsystem.py::OperatingSystemTestCase::test_positive_update_template PASSED [ 50%]
tests/foreman/api/test_template_combination.py::TemplateCombinationTestCase::test_positive_get_combination PASSED [ 60%]
tests/foreman/cli/test_location.py::LocationTestCase::test_positive_create_update_delete PASSED [ 70%]
tests/foreman/cli/test_location.py::LocationTestCase::test_positive_create_with_parent PASSED [ 80%]
tests/foreman/cli/test_operatingsystem.py::OperatingSystemTestCase::test_positive_add_template PASSED [ 90%]
tests/foreman/cli/test_organization.py::OrganizationTestCase::test_positive_add_and_remove_templates FAILED [100%]
===================== 2 failed, 8 passed in 169.81 seconds =====================
```

Failed tests: 
`test_positive_post_status_code` - automation bug (connected to bz workflow in robottelo not to provisioning template)
`test_positive_add_and_remove_templates` - BZ
